src/lib/mongoBackend/safeMongo.h

```diff
@@ -32,47 +32,57 @@
 
 #include "mongo/client/dbclient.h"
 
+/* Some macros to make the usage of these function prettier */
+#define getObjectFieldF(b, field)           getObjectField(b, field, __FUNCTION__, __LINE__)
+#define getStringFieldF(b, field)           getStringField(b, field, __FUNCTION__, __LINE__)
+#define getIntFieldF(b, field)              getIntField(b, field, __FUNCTION__, __LINE__)
+#define getLongFieldF(b, field)             getLongField(b, field, __FUNCTION__, __LINE__)
+#define getIntOrLongFieldAsLongF(b, field)  getIntOrLongFieldAsLong(b, field, __FUNCTION__, __LINE__)
+#define getBoolFieldF(b, field)             getBoolField(b, field, __FUNCTION__, __LINE__)
+#define getFieldF(b, field)                 getField(b, field, __FUNCTION__,  __LINE__)
+#define nextSafeOrErrorF(c, r, err)         nextSafeOrError(c, r, err, __FUNCTION__,  __LINE__)
+
 /* ****************************************************************************
 *
 * getObjectField -
 */
-extern mongo::BSONObj getObjectField(const mongo::BSONObj& b, const std::string& field);
+extern mongo::BSONObj getObjectField(const mongo::BSONObj& b, const std::string& field, const std::string& caller = "<none>", int line = 0);
```

Comment:
I don't like too much the default part for the parameters but I needed bo about changing thousand lines in the unit tests.

Maybe a "safer" alterantive could be:

```
#if UNIT_TEST
 extern mongo::BSONObj getObjectField(const mongo::BSONObj& b, const std::string& field);
 extern mongo::BSONObj getStringField(const mongo::BSONObj& b, const std::string& field);
  ...
#else
 extern mongo::BSONObj getObjectField(const mongo::BSONObj& b, const std::string& field, const::string& function, int line);
 extern mongo::BSONObj getStringField(const mongo::BSONObj& b, const std::string& field, const::string& function, int line);
  ...#endif
```

What do you think?